core/stream/convert.go

```diff
@@ -410,9 +471,29 @@ func ftou(count int, dst, src buf) error {
 		}
 	default:
 		scale := float64(dstMask)
+		min, max := float64(math.MaxFloat64), float64(0)
+		for i := 0; i < count; i++ {
+			f := math.Float64frombits(ss2.Read(64))
+			if f < min {
+				min = f
+			}
+			if f > max {
+				max = f
+			}
+			ss2.ReadPos += src.stride - 64
+		}
 		for i := 0; i < count; i++ {
 			f := float64(f64.FromBits(ss.Read(srcBits), srcExpBits, srcManBits))
 			if norm {
+				if src.component.Channel == Channel_Depth {
```

Comment:
The following lines of code are repeated several times in this PR. Isn't there a way to encapsulate them into a function?